src/test/opentrackIO_parser/python/opentrackIOlib.py

```diff
@@ -0,0 +1,304 @@
+        self.schema_str = schema_text           # the Sample raw JSON
+        self.verbose = verbose
+        self.schema_text = None             # the Schema raw JSON
+        self.trans_mult = 1.0                               # user-preferred units multiplier
+        self.rot_mult = 1.0
+        self.pd = None          # the parsed procotol dictionary
+        self.sd = None          # the parsed schema dictionary
+        self.sample_time_format = "sec"
+        self.sample_time_formats = ["sec","timecode","string"]
+        self.focus_dist_mult = 1.0
+
+    def Import_schema(self):
+        """Read the schema which governs the interpretation of the protocol"""
+        if self.schema_str:
+            try:
+                self.sd = json.loads(self.schema_str)
+            except json.decoder.JSONDecodeError:
+                print("Got JSONDecodeError while decoding the sample!")
+                self.sd = None
+            if not self.sd:
+                print("Import_schema(): Failed to parse OTIO schema file.")
+            else:                                   # we have a valid schema
+                print("Parsed the schema JSON successfully.")
+                if self.verbose:
+                    print("Contents of the parsed JSON schema dict:\n")
+                    print(self.sd)
+                    print("\n")
+        else:
+            print("ERROR: no schema provided!")
+            return -1
+
+    def Parse(self):        
+        """Ingest the text and store the JSON items in a dictionary"""
+        protocol = None
+        if self.sample_str != None:        
+            print("Parsing JSON string from sample buffer...")
+            #text = self.sample_str
+            #print(text)
+            #print('\n')    
+            try:
+                self.pd = json.loads(self.sample_str)
+            except:
+                print("Parse(): Error. Failed to parse JSON file! Either path incorrect or JSON invalid.")
+                exit(-1)
+            if not self.pd:
+                print("Parse(): Failed to parse OTIO sample file.")
+            else:                                   # we have a OTP message
+                print("Parsed the sample JSON successfully.")
+                if self.verbose:
+                    print("Contents of the parsed sample JSON dict:\n")
+                    print(self.pd)
+                    print("\n")
+        else:
+            print("Parse(): Error. No json text provided!")
+            return -1
+
+    def Conversion_factor_from_meters(self,unit_str):
+        """Stores the multiplicative conversion factor from meters to user-specified units
+        Valid unit specifier strings: m, cm, mm, in"""
+        if unit_str == "m":
+            return 1.0
+        elif unit_str == "cm":
+            return 100.0
+        elif unit_str == "mm":
+            return 1000.0
+        elif unit_str == "in":
+            return (1000/25.4)
```

Comment:
1000.0/25.4 to be sure it doesn't do an integer division

Reply:
Done